### ISSUE TITLE
getLocation() method never returns if service is enabled using code

### ISSUE THREAD

**Issue by junaidlodhi22** [Lyokone/flutterlocation#590]
if the location service is disabled, then the **getLocation()** method never returns if the services get enabled using code.

**Expected behavior**
it should return giving the current location of user

**Tested on:**
 - Android Nokia 6.2, API Level 30 real device

**Comment by abumoallim** [comment]
I am facing the same issue for a long time but unable to replicate the actual issue.

**Comment by abumoallim** [comment]
Did you find any solution for this?

**Comment by junaidlodhi22** [comment]
No, I did not find any solution for this yet.

**Comment by LostInDarkMath** [comment]
Same issue here.

**Comment by junaidlodhi22** [comment]
we can put a timeout call for it to return always. Workaround maybe

**Comment by luckv** [comment]
I also has this issue. 
When the user don't activate the fused location provider service (press "*no,thanks*" in the shown dialog), and `isServiceEnabled()` returns `false`,`getLocation()` hangs and never returns. 
This is a problem I encountered in different devices and android versions.
Some devices/os I tried:
 - Realme 7, android 10
 - Samsung galaxy A5 (2017), android 8.0.0
 - Samsung galaxy S10e, android 11
 - Emulator with google play store with android 10,11,12

It's replicable, just respond no to the dialog of fused location provider (or never call `requestService()`) and then call `getLocation()`
